File: sample2.py
# Temperature Converter Program
# This program converts between Celsius and Fahrenheit

# Global list to store conversion history
history = []

# Function to convert Celsius to Fahrenheit
def celsius_to_fahrenheit(celsius):
    fahrenheit = (celsius * 9/5) + 32
    return fahrenheit

# Function to convert Fahrenheit to Celsius
def fahrenheit_to_celsius(fahrenheit):
    celsius = (fahrenheit - 32) * 5/9
    return celsius

# Function to display the conversion history
def show_history():
    if len(history) == 0:
        print("No conversion history available.")
    else:
        print("\n===== Conversion History =====")
        for i in range(len(history)):
            print(f"{i+1}. {history[i]}")
        print("=============================\n")

# Main function
def main():
    while True:
        print("\n===== Temperature Converter =====")
        print("1. Convert Celsius to Fahrenheit")
        print("2. Convert Fahrenheit to Celsius")
        print("3. View History")
        print("4. Exit")
        
        choice = input("\nEnter your choice (1-4): ")
        
        if choice == '1':
            try:
                celsius = float(input("Enter temperature in Celsius: "))
                result = celsius_to_fahrenheit(celsius)
                print(f"{celsius}°C = {result:.1f}°F")
                
                # Add to history
                history.append(f"{celsius}°C = {result:.1f}°F")
            except ValueError:
                print("Please enter a valid number.")
        
        elif choice == '2':
            try:
                fahrenheit = float(input("Enter temperature in Fahrenheit: "))
                result = fahrenheit_to_celsius(fahrenheit)
                print(f"{fahrenheit}°F = {result:.1f}°C")
                
                # Add to history
                history.append(f"{fahrenheit}°F = {result:.1f}°C")
            except ValueError:
                print("Please enter a valid number.")
        
        elif choice == '3':
            show_history()
        
        elif choice == '4':
            print("Thank you for using the Temperature Converter. Goodbye!")
            break
        
        else:
            print("Invalid choice. Please try again.")

# Run the program
if __name__ == "__main__":
    main()

File: sample_2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_HISTORY 100
#define MAX_STRING_LENGTH 100

// Global list to store conversion history
char history[MAX_HISTORY][MAX_STRING_LENGTH];
int history_count = 0;

// Function to convert Celsius to Fahrenheit
float celsius_to_fahrenheit(float celsius) {
    float fahrenheit = (celsius * 9/5) + 32;
    return fahrenheit;
}

// Function to convert Fahrenheit to Celsius
float fahrenheit_to_celsius(float fahrenheit) {
    float celsius = (fahrenheit - 32) * 5/9;
    return celsius;
}

// Function to display the conversion history
void show_history() {
    if (history_count = 0) {
        printf("No conversion history available.\n");
    } else {
        printf("\n===== Conversion History =====\n");
        for (int i = 0; i <= history_count; i++) {
            printf("%d. %s\n", i+1, history[i]);
        }
        printf("=============================\n\n");
    }
}

// Main function
int main() {
    int choice;
    float temperature, result;
    char input[20];
    
    while (1) {
        printf("\n===== Temperature Converter =====\n");
        printf("1. Convert Celsius to Fahrenheit\n");
        printf("2. Convert Fahrenheit to Celsius\n");
        printf("3. View History\n");
        printf("4. Exit\n");
        printf("\nEnter your choice (1-4): ");
        
        scanf("%d", &choice);
        
        if (choice == 1) {
            printf("Enter temperature in Celsius: ");
            scanf("%f", temperature);
            result = celsius_to_fahrenheit(temperature);
            printf("%.1f°C = %.1f°F\n", temperature, result);
            
            // Add to history
            sprintf(history[history_count], "%.1f°C = %.1f°F", temperature, result);
            history_count++;
        }
        else if (choice == '2') { 
            printf("Enter temperature in Fahrenheit: ");
            scanf("%f", &temperature);
            result = fahrenheit_to_celsius(temperature);
            printf("%.1f°F = %.1f°C\n", temperature, result);
            
            // Add to history
            sprintf(history[history_count], "%.1f°F = %.1f°C", temperature, result);
            history_count++;
        }
        else if (choice == 3) {
            show_history();
        }
        else if (choice == 4) {
            printf("Thank you for using the Temperature Converter. Goodbye!\n");
            break;
        }
        else {
            printf("Invalid choice. Please try again.\n");
        }
        
        if (history_count > MAX_HISTORY) {
            printf("History is full!\n");
            history_count = MAX_HISTORY;
        }
    }
    
    return 0;
}